llama-index-integrations/vector_stores/llama-index-vector-stores-opensearch/llama_index/vector_stores/opensearch/base.py

```diff
@@ -388,13 +388,12 @@ async def aquery(
             )
             params = {
                 "search_pipeline": self._search_pipeline,
-                "_source_excludes": ["embedding"],
             }
         else:
             search_query = self._knn_search_query(
                 self._embedding_field, query_embedding, k, filters=filters
             )
-            params = {"_source_excludes": ["embedding"]}
```

Comment:
WDYT @dosubot

Reply:
I think there's definitely a use-case for keeping the embeddings on the response nodes (i.e. visualization, debugging later, etc.)

Maybe we can add a config to the class constructor? (imo it would be nice if every vector store gave this option lol)

Reply:
Every [query](https://github.com/run-llama/llama_index/blob/3113bec2b119f38883886e0ec8a0473456bafc39/llama-index-core/llama_index/core/vector_stores/types.py#L398) (or aquery) method has a query parameter (of type `VectorStoreQuery`), thus I think the `query.output_fields` might be helpful in this scenario:

https://github.com/run-llama/llama_index/blob/3113bec2b119f38883886e0ec8a0473456bafc39/llama-index-core/llama_index/core/vector_stores/types.py#L223

Reply:
Thanks @logan-markewich 

I'll revert the exclusion then.

As for the option, I agree with @RussellLuo, maybe better to put that in VectorStoreQuery. I also think other optional use cases like pagination or something like that.

The problem is, that the upper layer doesn't offer that flexibility, like query engine and retriever.
```
class BaseQueryEngine(ChainableMixin, PromptMixin):
    """Base query engine."""
   def query(self, str_or_query_bundle: QueryType) -> RESPONSE_TYPE:
         # Logic

class BaseRetriever(ChainableMixin, PromptMixin):
    """Base retriever."""

    @dispatcher.span
    def retrieve(self, str_or_query_bundle: QueryType) -> List[NodeWithScore]:
         # Logic
```

So I'll leave that to separate PR

Reply:
Yea the upper-layers don't really expose VectorStoreQuery well (its quite hidden), so setting `output_fields` can be a little janky (There is a `vector_store_kwargs` thing you can pass in when creating the index or retriever, and I think that eventually makes its way to the vector query object)

Reply:
well actually, I don't even think the embeddings are being attached to the returned nodes anyways right now too